gptel-context.el

```diff
@@ -163,32 +165,57 @@ context chunk.  This is accessible as, for example:
         (eq buffer-file-coding-system 'no-conversion))
     (file-missing (message "File \"%s\" is not readable." path))))
 
+(defun gptel-context--handle-binary (path)
+  "Add binary file at PATH to context if supported.
+Return PATH if added, nil if ignored."
+  (if-let* (((gptel--model-capable-p 'media))
+            (mime (mailcap-file-name-to-mime-type path))
+            ((gptel--model-mime-capable-p mime)))
+      (prog1 path
+        (cl-pushnew (list path :mime mime)
+                    gptel-context--alist :test #'equal)
+        (message "File \"%s\" added to context." path))
+    (message "Ignoring unsupported binary file \"%s\"." path)
+    nil))
+
+(defun gptel-context--handle-directory (path action)
+  "Process all files in directory at PATH according to ACTION.
+ACTION should be either `add' or `remove'."
+  (let ((files (directory-files-recursively path "." t)))
+    (mapc (lambda (file)
+            (unless (file-directory-p file)
+              (pcase action
+		('add (gptel--file-binary-p file)
+		      (gptel-context--handle-binary file)
+		      (cl-pushnew (list file) gptel-context--alist :test #'equal)
+		      (message "File \"%s\" added to context." file))
+		('remove
+		 (setf (alist-get file gptel-context--alist nil 'remove #'equal) nil)
+		 (message "File \"%s\" removed from context." file)))))
+          files)))
+
 (defun gptel-context-add-file (path)
   "Add the file at PATH to the gptel context.
-
+If PATH is a directory, recursively add all files in it.
 PATH should be readable as text."
   (interactive "fChoose file to add to context: ")
-  (if (gptel--file-binary-p path)   ;Attach if supported
-      (if-let* (((gptel--model-capable-p 'media))
-                (mime (mailcap-file-name-to-mime-type path))
-                ((gptel--model-mime-capable-p mime)))
-          (prog1 path
-            (cl-pushnew (list path :mime mime)
-                        gptel-context--alist :test #'equal)
-            (message "File \"%s\" added to context." path))
-        (message "Ignoring unsupported binary file \"%s\"." path))
-    ;; Add text file
-    (cl-pushnew (list path) gptel-context--alist :test #'equal)
-    (message "File \"%s\" added to context." path)
-    path))
+  (if (file-directory-p path)
```

Comment:
Convert to a `cond` clause, it's easier to read here.

Reply:
Addressed: https://github.com/karthink/gptel/pull/438/commits/bc9138a525120e8786d7390a436e44895e660c09